### PR TITLE
Preneeds attachment upload

### PR DESCRIPTION
**Pull request by lihanli**
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/5936

### CODE REVIEW

**Comment by jbalboni** [comment]
@lihanli Am I understanding this correctly that the front end would be sending the attachments in a `preneed_attachment_guids` field in the response, at the same level as the applicant, claimant, and veteran info? And we'd need to update the schema to accomodate this?

File: app/controllers/v0/preneeds/preneed_attachments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+module V0
+  module Preneeds
+    class PreneedAttachmentsController < PreneedsController
+      def create
+        preneed_attachment = ::Preneeds::PreneedAttachment.new
+        preneed_attachment.set_file_data!(params[:preneed_attachment][:file_data])
```
Comment (jbalboni) [inline app/controllers/v0/preneeds/preneed_attachments_controller.rb]: Can this match what we're doing for pension/burials, as far as params being passed in and data being sent back? (It looks different, though I could be wrong)

Comment (lihanli) [reply app/controllers/v0/preneeds/preneed_attachments_controller.rb]: this controller matches the education benefits claim controller and structures the params in the standard rails way, the other controller is non standard and later we should change it to match this one

**Comment by lihanli** [comment]
@jbalboni i think we should add `preneed_attachment_guids` to the schema and remove the attachments field. that field is used for submitting to their api but frontend doesn't need to know about it.

**Comment by jbalboni** [comment]
@lihanli Ok, that's fine. I also see in the mock ups that there's a document category that users can set. Is that something you're expecting on the backend during the final submission?

**Comment by lihanli** [comment]
yea i'm going to add attachment_id to this controller, which the frontend will receive from the /v0/preneeds/attachment_types request

**Comment by jbalboni** [comment]
@lihanli Do you mean the attachments controller? Of the controller for the preneed form submission? I would prefer not to make requests to the backend whenever a user changes that field after they've uploaded. It would be easier if I could include it in the `preneed_attachments_guids` array in the submission.

**Comment by lihanli** [comment]
that would work too, we can rename it to preneed_attachments  and have a hash with guid and attachment_id

**Comment by lihanli** [comment]
@rlbaker hey this is ready for review, api part is in another branch and not working but we can test file uploads to s3 in this branch

**Comment by rlbaker** [comment]
Sounds good. Giving everything another look now.

**Pull request merged**